src/helpers/signingWallet.ts

```diff
@@ -0,0 +1,118 @@
+import { verifyMessage, Wallet } from '@ethersproject/wallet';
+import lang from 'i18n-js';
+// @ts-ignore
+import { RAINBOW_MASTER_KEY } from 'react-native-dotenv';
+import { loadString, saveString } from '../model/keychain';
+import * as keychain from '../model/keychain';
+import { PrivateKeyData, publicAccessControlOptions } from '../model/wallet';
+import { privateKeyKey } from '../utils/keychainConstants';
+import AesEncryptor from '@rainbow-me/handlers/aesEncryption';
+import { logger } from '@rainbow-me/utils';
+
+const authenticationPrompt = lang.t('wallet.authenticate.please');
+
+export async function getPublicKeyOfTheSigningWalletAndCreateWalletIfNeeded(): Promise<string> {
+  let alreadyExistingWallet = await loadString('signing_wallet_address');
+  if (typeof alreadyExistingWallet !== 'string') {
+    const wallet = Wallet.createRandom();
+    logger.log('Created signing wallet');
+    const { privateKey, address } = wallet;
+    const encryptor = new AesEncryptor();
+    const encryptedPrivateKey = (await encryptor.encrypt(
+      RAINBOW_MASTER_KEY,
+      privateKey
+    )) as string;
+
+    await saveString(
+      'signing_wallet',
```

Comment:
can we pull these keys `signing_wallet` and `signing_wallet_address` out to separate constants or add them to keychainConstants file